benchmarks/nemo/nemolite2d/manual_versions/psykal_kokkos/time_step_views_kokkos.cpp

```diff
@@ -0,0 +1,616 @@
+    double_2dview dx_u_view("dx_u", internal_xstop+1, internal_ystop+1);
+    double_2dview dx_v_view("dx_v", internal_xstop+1, internal_ystop+1);
+    double_2dview dx_t_view("dx_t", internal_xstop+1, internal_ystop+1);
+    double_2dview dy_u_view("dy_u", internal_xstop+1, internal_ystop+1);
+    double_2dview dy_v_view("dy_v", internal_xstop+1, internal_ystop+1);
+    double_2dview dy_t_view("dy_t", internal_xstop+1, internal_ystop+1);
+    double_2dview gphiu_view("gphiu", internal_xstop+1, internal_ystop+1);
+    double_2dview gphiv_view("gphiv", internal_xstop+1, internal_ystop+1);
+
+
+    // Create Mirrors (this overlap thew views if the execution device is the host)
+    double_2dview::HostMirror h_ssha_t = Kokkos::create_mirror_view( ssha_t_view );
+    double_2dview::HostMirror h_sshn_t = Kokkos::create_mirror_view( sshn_t_view );
+    double_2dview::HostMirror h_sshn_u = Kokkos::create_mirror_view( sshn_u_view );
+    double_2dview::HostMirror h_sshn_v = Kokkos::create_mirror_view( sshn_v_view );
+    double_2dview::HostMirror h_hu = Kokkos::create_mirror_view( hu_view );
+    double_2dview::HostMirror h_hv = Kokkos::create_mirror_view( hv_view );
+    double_2dview::HostMirror h_un = Kokkos::create_mirror_view( un_view );
+    double_2dview::HostMirror h_vn = Kokkos::create_mirror_view( vn_view );
+    double_2dview::HostMirror h_ua = Kokkos::create_mirror_view( ua_view );
+    double_2dview::HostMirror h_ht = Kokkos::create_mirror_view( ht_view );
+    double_2dview::HostMirror h_ssha_u = Kokkos::create_mirror_view( ssha_u_view );
+    double_2dview::HostMirror h_va = Kokkos::create_mirror_view( va_view );
+    double_2dview::HostMirror h_ssha_v = Kokkos::create_mirror_view( ssha_v_view );
+
+    int_2dview::HostMirror h_tmask = Kokkos::create_mirror_view( tmask_view );
+    double_2dview::HostMirror h_area_t = Kokkos::create_mirror_view( area_t_view );
+    double_2dview::HostMirror h_area_u = Kokkos::create_mirror_view( area_u_view );
+    double_2dview::HostMirror h_area_v = Kokkos::create_mirror_view( area_v_view );
+    double_2dview::HostMirror h_dx_u = Kokkos::create_mirror_view( dx_u_view );
+    double_2dview::HostMirror h_dx_v = Kokkos::create_mirror_view( dx_v_view );
+    double_2dview::HostMirror h_dx_t = Kokkos::create_mirror_view( dx_t_view );
+    double_2dview::HostMirror h_dy_u = Kokkos::create_mirror_view( dy_u_view );
+    double_2dview::HostMirror h_dy_v = Kokkos::create_mirror_view( dy_v_view );
+    double_2dview::HostMirror h_dy_t = Kokkos::create_mirror_view( dy_t_view );
+    double_2dview::HostMirror h_gphiu = Kokkos::create_mirror_view( gphiu_view );
+    double_2dview::HostMirror h_gphiv = Kokkos::create_mirror_view( gphiv_view );
+
+
+    // Copy Fortran arrays into the Kokkos View Mirrors
+    for(int jj=0; jj < internal_ystop+1; jj++){
+        for(int ji=0; ji < internal_xstop+1; ji++){
+            int idx = jj*width + ji;
+            h_ssha_t(jj, ji) = ssha_t[idx];
```

Comment:
So a Kokkos 'View' gives us back Fortran array notation? :-)

Reply:
Not only that, you can even instantiate the View template with a `LayoutLeft` parameter and then use (ji, jj) as in the Fortran implementation. I chose not to do that :)